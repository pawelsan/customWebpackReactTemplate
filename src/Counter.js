import React, { useState } from "react";

const Counter = () => {
  const [counter, setCounter] = useState(0);
  return (
    <div>
      <h2>A counter</h2>
      <p>{counter}</p>

      <button type="button" onClick={() => setCounter(counter + 1)}>
        Increment
      </button>

      <button type="button" onClick={() => setCounter(counter - 1)}>
        Decrement
      </button>
    </div>
  );
};

export default Counter;
