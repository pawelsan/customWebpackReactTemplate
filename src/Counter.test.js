// describe("My Test Suite", () => {
//   it("My Test Case", () => {
//     expect(true).toEqual(true);
//   });
// });

import React from "react";
import renderer from "react-test-renderer";

import Counter from "./Counter";

describe("Counter", () => {
  test("snapshot renders", () => {
    const component = renderer.create(<Counter counter={1} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
