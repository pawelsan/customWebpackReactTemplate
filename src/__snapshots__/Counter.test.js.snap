// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot renders 1`] = `
<div>
  <h2>
    A counter
  </h2>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
    type="button"
  >
    Increment
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Decrement
  </button>
</div>
`;
